### DISCUSSION TITLE
Center first couple windows in a workspace

### DISCUSSION THREAD

**Discussion by kra-mo** [YaLTeR/niri#287] [Ideas]
I would be great to have the first couple windows in a workspace that don't yet take up more width than the display to be horizontally centered instead of tiling them from the left.

Basically, `on-overflow` but before windows actually overflow.

**Comment by YaLTeR** [comment]
I was thinking about how to do this (also PaperWM does this I believe). Essentially the question I've hit is that:

- If we do it always, then it's impossible to have 1 window flush to the side as is occasionally convenient (or maybe it's not such a big problem?).
- If we want to somehow do it "only when windows are first spawned": I couldn't come up with reasonable logic here that would always do what you want/expect. I.e. it either won't re-center windows again when they shrink from overflowing to not overflowing, which seems desirable.
